app/src/full/java/io/homeassistant/companion/android/sensors/AndroidAutoSensorManager.kt

```diff
@@ -70,11 +68,11 @@ class AndroidAutoSensorManager : SensorManager, Observer<Int> {
     }
 
     override fun onChanged(value: Int) {
-        if (!isEnabled(context, androidAutoConnected)) {
-            CoroutineScope(Dispatchers.Main + Job()).launch {
+        sensorWorkerScope.launch {
```

Comment:
This check will now run async and the code below will continue even if the observer is removed inside this job. I think you should move everything inside the `launch`.

Reply:
done